tests/jobs/test_scheduler_job.py

```diff
@@ -1531,6 +1531,50 @@ def test_adopt_or_reset_orphaned_tasks(self, dag_maker):
         ti2 = dr2.get_task_instance(task_id=op1.task_id, session=session)
         assert ti2.state == State.QUEUED, "Tasks run by Backfill Jobs should not be reset"
 
+    def test_fail_stuck_queued_tasks(self, dag_maker):
+        session = settings.Session()
+        with dag_maker("test_fail_stuck_queued_tasks"):
+            op1 = EmptyOperator(task_id="op1")
+
+        dr = dag_maker.create_dagrun()
+        ti = dr.get_task_instance(task_id=op1.task_id, session=session)
+        ti.state = State.QUEUED
+        ti.queued_dttm = timezone.utcnow() - timedelta(minutes=15)
+        session.commit()
+
+        processor = mock.MagicMock()
+
+        self.scheduler_job = SchedulerJob(num_runs=0)
+        self.scheduler_job.processor_agent = processor
+        self.scheduler_job._task_queued_timeout = 300
+
+        self.scheduler_job._fail_tasks_stuck_in_queued()
```

Comment:
Should we mock `cleanup_stuck_queued_tasks` and make sure it is called with the TI?